### PR TITLE
Query nova client for allowed flavors based on config

### PR DESCRIPTION
**Pull request by heytrav**
* Set min RAM and vcpu for flavor
* tests

### CODE REVIEW

File: magnum_capi_helm/driver.py [hunk 1]
```diff
@@ -572,6 +572,34 @@ def _get_fixed_network_id(self, context, cluster):
                 context, network, source="name", target="id", external=False
             )
 
+    def _validate_allowed_flavor(self, context, requested_flavor):
+        # Compare requested flavor with allowed for Kubernetes node
+        flavors = (
+            clients.OpenStackClients(context)
+            .nova()
+            .flavors.list(min_ram=CONF.capi_helm.minimum_flavor_ram)
```
Comment (JohnGarbutt) [inline magnum_capi_helm/driver.py:580]: I feel there must be a more efficient way, but this is less API requests than looking up by name and id, so maybe this is the best option.

**Review by JohnGarbutt — APPROVED** [review]
Very good add. For sure it breaks with less than 2 vCPUs, as kubelet fails to start.

**Comment by mkjpryor** [comment]
Looks good to me, subject to test suite

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**